Sonar quality enhancement - Batch 1

Chenjp opened the pull request:
Sonar quality enhancement.

## What is the purpose of the change

SonarCloud code quality enhancement.

## Brief changelog


## Verifying this change


<!-- Follow this checklist to help us incorporate your contribution quickly and easily: -->

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

Chenjp committed:
Sonar quality enhancement - Batch 1

Sonar quality enhancement.

AlbumenJ commented on dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java:
```diff
@@ -124,19 +110,16 @@ public <T> T getOrRegisterBean(Class<T> type) {
-                if (bean == null) {
-                    bean = createAndRegisterBean(name, type);
-                }
+    private static final Hashtable<Class<?>, ReentrantLock> typeLocks = new Hashtable<>();
```
Why not use `ConcurrentHashMap`?

Chenjp replied:
new commit: replace hashtable with concurrenthashmap.

Chenjp committed:
Update ScopeBeanFactory.java

Follows AlbumenJ suggestion, change hashtable to concurrenthashmap to improve read performance.

AlbumenJ commented on dubbo-common/src/main/java/org/apache/dubbo/common/config/ConfigurationUtils.java:
```diff
@@ -280,9 +280,7 @@ private static <V extends Object> Map<String, V> getSubProperties(
 
         if (CollectionUtils.isNotEmptyMap(configMap)) {
             Map<String, V> copy;
-            synchronized (configMap) {
```
Why remove this sync

AlbumenJ commented on dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java:
```diff
@@ -145,11 +145,19 @@ public void dispatchThreadPoolExhaustedEvent(String msg) {
         listeners.forEach(listener -> listener.onEvent(new ThreadPoolExhaustedEvent(msg)));
     }
 
+    public static long getLastPrintTime() {
+        return lastPrintTime;
+    }
+
+    public static void setLastPrintTime(long lastPrintTime) {
+        AbortPolicyWithReport.lastPrintTime = lastPrintTime;
+    }
+
```
meanless here

AlbumenJ commented on dubbo-compatible/src/main/java/com/alibaba/dubbo/container/page/PageServlet.java:
```diff
@@ -56,14 +56,18 @@ public static PageServlet getInstance() {
         return INSTANCE;
     }
 
+    private static void setInstance(PageServlet iNSTANCE) {
+        INSTANCE = iNSTANCE;
+    }
+
```
Still meanless

oxsean commented:
I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.

Chenjp commented:
> I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.

How about to disable those meaningless rules in sonar admin?

Chenjp commented:
This PR is not a bug fix.
Since sonar quality rule is a nice-to-have item. Close the PR.

oxsean commented:
> > I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.
> 
> 
> 
> How about to disable those meaningless rules in sonar admin?

I don't know much about it either. If you have time, could you look into it?
